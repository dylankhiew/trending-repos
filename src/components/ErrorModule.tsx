import {
  Image,
  ImageSourcePropType,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import { COLOR } from "../constants/colorConstants";

interface ErrorModuleProps {
  title: string;
  subtitle: string;
  buttonTitle: string;
  imageSource: ImageSourcePropType;
  onPress: () => void;
}

export default function ErrorModule({
  onPress,
  title,
  subtitle,
  buttonTitle,
  imageSource,
}: ErrorModuleProps): JSX.Element {
  return (
    <SafeAreaView style={styles.container}>
      <Image
        resizeMode="contain"
        style={styles.errorImage}
        source={imageSource}
      />
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.subtitle}>{subtitle}</Text>
      <View style={styles.buttonContainer}>
        <TouchableOpacity style={styles.button} onPress={onPress}>
          <Text style={styles.buttonText}>{buttonTitle}</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 48,
  },
  errorImage: {
    height: 300,
  },
  title: {
    marginTop: 24,
    fontSize: 18,
    fontFamily: "Roboto-Bold",
  },
  subtitle: {
    marginTop: 8,
    fontSize: 15,
    fontFamily: "Roboto-Regular",
    color: COLOR.GRAY200,
  },
  buttonContainer: {
    width: "100%",
    position: "absolute",
    bottom: 32,
    height: 48,
  },
  button: {
    borderColor: COLOR.GREEN,
    backgroundColor: COLOR.WHITE,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    heigth: 48,
    marginHorizontal: 16,
    borderWidth: 1,
    borderRadius: 4,
    flex: 1,
  },
  buttonText: {
    fontFamily: "Roboto-Medium",
    fontSize: 15,
    color: COLOR.GREEN,
    textTransform: "uppercase",
  },
});
