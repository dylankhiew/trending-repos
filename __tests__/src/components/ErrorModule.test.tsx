import React from 'react';
import renderer from 'react-test-renderer';

import ErrorModule from '../../../src/components/ErrorModule';

it('should render correctly', () => {
  const title = 'title';
  const subtitle = 'subtitle';
  const buttonTitle = 'buttonTitle';

  const tree = renderer
    .create(
      <ErrorModule
        title={title}
        subtitle={subtitle}
        buttonTitle={buttonTitle}
        imageSource={0}
        onPress={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
