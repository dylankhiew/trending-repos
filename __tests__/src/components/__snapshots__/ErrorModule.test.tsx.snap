// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 48,
    }
  }
>
  <Image
    resizeMode="contain"
    source={0}
    style={
      {
        "height": 300,
      }
    }
  />
  <Text
    style={
      {
        "color": "#4A4A4A",
        "fontFamily": "Roboto-Bold",
        "fontSize": 18,
        "marginTop": 24,
      }
    }
  >
    title
  </Text>
  <Text
    style={
      {
        "color": "#929292",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 8,
      }
    }
  >
    subtitle
  </Text>
  <View
    style={
      {
        "bottom": 32,
        "height": 48,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#31B057",
          "borderRadius": 4,
          "borderWidth": 1,
          "display": "flex",
          "flex": 1,
          "heigth": 48,
          "justifyContent": "center",
          "marginHorizontal": 16,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#31B057",
            "fontFamily": "Roboto-Medium",
            "fontSize": 15,
            "textTransform": "uppercase",
          }
        }
      >
        buttonTitle
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
